drivers/serial/uart_nrfx_uart.c

```diff
@@ -24,39 +25,30 @@
 #define HAS_PROP(prop)	DT_INST_NODE_HAS_PROP(0, prop)
 
 #define BAUDRATE	PROP(current_speed)
-#define TX_PIN		PROP(tx_pin)
 
-#define RX_PIN_USED	HAS_PROP(rx_pin)
-#if RX_PIN_USED
-#define RX_PIN		PROP(rx_pin)
-#else
-#define RX_PIN		NRF_UART_PSEL_DISCONNECTED
-#endif
-
-#define HW_FLOW_CONTROL_AVAILABLE	(HAS_PROP(rts_pin) || HAS_PROP(cts_pin))
-
-/* Protect against enabling flow control without pins set. */
-BUILD_ASSERT((PROP(hw_flow_control) && HW_FLOW_CONTROL_AVAILABLE) ||
-		!PROP(hw_flow_control));
-
-#define RTS_PIN \
-	COND_CODE_1(HAS_PROP(rts_pin), \
-		(PROP(rts_pin)), (NRF_UART_PSEL_DISCONNECTED))
-
-#define CTS_PIN \
-	COND_CODE_1(HAS_PROP(cts_pin), \
-		(PROP(cts_pin)), (NRF_UART_PSEL_DISCONNECTED))
+#define DISABLE_RX	HAS_PROP(disable_rx)
+#define HW_FLOW_CONTROL_AVAILABLE	HAS_PROP(hw_flow_control)
```

Comment:
what is your suggestion here? Should we add a new DT property as we do for `disable_rx`?

Reply:
I think we should get rid of the macro and make the flow control always supported, event though when both RTS and CTS pins are not connected, it will be actually unusable. Otherwise, we would need to check at runtime if these pins are connected, by looking into their PSEL registers, right? I don't think it is worth it.
And similar approach will need to be applied in uart_nrfx_uarte.c (to get rid of `UARTE_CFG_FLAG_*_PIN_SET` flags and related macros).